projects/media-viewer/src/lib/comments-summary/comments-summary.component.ts

```diff
@@ -1,31 +1,67 @@
-import { Component, Input, ViewChild, ElementRef } from '@angular/core';
-import { Subject } from 'rxjs';
-import { Comment } from '../annotations/comment-set/comment/comment.model';
+import { Component, Input, ViewChild, ElementRef, OnChanges } from '@angular/core';
 import { PrintService } from '../print.service';
+import { ToolbarEventService } from '../toolbar/toolbar-event.service';
+import { AnnotationSet } from '../annotations/annotation-set/annotation-set.model';
+import { CommentsSummary } from './comments-summary.model';
 
 @Component({
   selector: 'mv-comments-summary',
   templateUrl: './comments-summary.component.html',
   styleUrls: ['./comments-summary.component.scss']
 })
-export class CommentsSummaryComponent {
+export class CommentsSummaryComponent implements OnChanges {
 
-  @Input() showCommentSummary: Subject<boolean>;
-  @Input() comments: Comment[];
   @Input() title: string;
+  @Input() contentType: string;
+  @Input() annotationSet: AnnotationSet | null;
+
+  comments: CommentsSummary[] = [];
 
   @ViewChild('commentContainer') commentsTable: ElementRef;
 
   constructor(
-    private readonly printService: PrintService
+    private readonly printService: PrintService,
+    private readonly toolbarEvents: ToolbarEventService
   ) {}
 
+  ngOnChanges() {
+    if (this.annotationSet) {
+      this.generateCommentsSummary();
+      this.orderCommentsSummary();
+    }
+  }
+
+  generateCommentsSummary() {
+    this.annotationSet.annotations
+      .forEach(c => {
+        this.comments.push({
+          page: c.page,
+          comment: c.comments[0],
+          x: c.rectangles[0].x,
+          y: c.rectangles[0].y
+        });
+      });
+  }
+
+  orderCommentsSummary() {
+    this.comments
+      .sort((a, b) => a.x >= b.x ? 1 : -1)
+      .sort((a, b) => a.y >= b.y ? 1 : -1)
+      .sort((a, b) => a.page - b.page);
+  }
+
   public onClose(): void {
-    this.showCommentSummary.next(false);
+    this.toolbarEvents.displayCommentSummary();
   }
 
   public onPrint(): void {
     this.printService.printElementNatively(this.commentsTable.nativeElement, 1000, 900);
   }
 
+  navigateToPage(pageNumber: number) {
+    if (this.contentType === 'pdf') {
+      this.toolbarEvents.setPage(pageNumber);
+    }
+    this.toolbarEvents.displayCommentSummary();
```

Comment:
with talebs new toolbar, does this open the comment panel if its closed, otherwise all this does is close the summary right?